scenarios/java/openjdk/captioning/Caption.java

```diff
@@ -0,0 +1,423 @@
+
+final class UserConfig
+{
+    final private boolean profanityFilterRemoveEnabled;
+    final private boolean profanityFilterMaskEnabled;
+    final private Optional<String> inputFile;
+    final private Optional<String> outputFile;
+    final private Optional<String> phraseList;
+    final private boolean suppressOutputEnabled;
+    final private boolean partialResultsEnabled;
+    final private Optional<String> stablePartialResultThreshold;
+    final private boolean srtEnabled;
+    final private boolean trueTextEnabled;
+    final private String subscriptionKey;
+    final private String region;
+
+    public boolean getProfanityFilterRemoveEnabled() {
+        return profanityFilterRemoveEnabled;
+    }
+    public boolean getProfanityFilterMaskEnabled() {
+        return profanityFilterMaskEnabled;
+    }
+    public Optional<String> getInputFile() {
+        return inputFile;
+    }
+    public Optional<String> getOutputFile() {
+        return outputFile;
+    }
+    public Optional<String> getPhraseList() {
+        return phraseList;
+    }
+    public boolean getSuppressOutputEnabled() {
+        return suppressOutputEnabled;
+    }
+    public boolean getPartialResultsEnabled() {
+        return partialResultsEnabled;
+    }
+    public Optional<String> getStablePartialResultThreshold() {
+        return stablePartialResultThreshold;
+    }
+    public boolean getSrtEnabled() {
+        return srtEnabled;
+    }
+    public boolean getTrueTextEnabled() {
+        return trueTextEnabled;
+    }
+    public String getSubscriptionKey() {
+        return subscriptionKey;
+    }
+    public String getRegion() {
+        return region;
+    }
+
+    public UserConfig(
+        boolean profanityFilterRemoveEnabled,
+        boolean profanityFilterMaskEnabled,
+        Optional<String> inputFile,
+        Optional<String> outputFile,
+        Optional<String> phraseList,
+        boolean suppressOutputEnabled,
+        boolean partialResultsEnabled,
+        Optional<String> stablePartialResultThreshold,
+        boolean srtEnabled,
+        boolean trueTextEnabled,
+        String subscriptionKey,
+        String region
+        )
+    {
+        this.profanityFilterRemoveEnabled = profanityFilterRemoveEnabled;
+        this.profanityFilterMaskEnabled = profanityFilterMaskEnabled;
+        this.inputFile = inputFile;
+        this.outputFile = outputFile;
+        this.phraseList = phraseList;
+        this.suppressOutputEnabled = suppressOutputEnabled;
+        this.partialResultsEnabled = partialResultsEnabled;
+        this.stablePartialResultThreshold = stablePartialResultThreshold;
+        this.srtEnabled = srtEnabled;
+        this.trueTextEnabled = trueTextEnabled;
+        this.subscriptionKey = subscriptionKey;
+        this.region = region;
+    }
+}
+
+public class Caption
+{
+/*
+Helper functions
+*/
+
+    static Optional<String> GetCmdOption(List<String> args, String option)
```

Comment:
>GetCmdOption [](http://example.com/codeflow?start=28&length=12)

In Java method names start with lowercase, that has been practice in other Java samples